packages/flutter/lib/src/material/dialog.dart

```diff
@@ -961,38 +961,69 @@ Widget _buildMaterialDialogTransitions(BuildContext context, Animation<double> a
 ///  * <https://material.io/design/components/dialogs.html>
 Future<T?> showDialog<T>({
```

Comment:
The documentation for all the show*Dialog methods should have a section about state restoration and how using these methods will not enable state restoration for the dialog. Ideally, we would also include a code sample that shows how to push a restorable dialog.